google/cloud/bigquery/_job_helpers.py

```diff
@@ -156,7 +168,12 @@ def do_query():
         else:
             return query_job
 
-    future = do_query()
+    # Allow folks to turn off job_retry with an explicit None.
+    if job_retry is None:
+        future = do_query()
+    else:
+        future = job_retry(do_query)()
```

Comment:
Note: even though I add the retry here, we'll still need the one in `QueryJob`, because BigQuery queries can be started successfully but still fail for a retriablle reason. By adding the retry here, we can retry the queries that fail for a retriable reason.